sdk/react/src/lib/fragment-components/a-select.tsx

```diff
@@ -0,0 +1,566 @@
+	render: ({ componentObj, select, remove, selected }) => <ASelect
+		componentObj={componentObj}
+		select={select}
+		remove={remove}
+		selected={selected}>
+	</ASelect>,
+	keywords: ['select'],
+	name: 'Select',
+	type: 'select',
+	defaultComponentObj: {
+		type: 'select',
+		labelText: 'Select',
+		items: [
+			{
+				text: 'Option 1',
+				value: 'placeholder-item',
+				type: 'select-item'
+			}
+		]
+	},
+	image,
+	codeExport: {
+		angular: {
+			latest: {
+				inputs: ({ json }) => `@Input() ${nameStringToVariableString(json.codeContext?.name)}Disabled = ${json.disabled ?? false};
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}Invalid = ${json.invalid ?? false};
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}Warn = ${json.warn ?? false};
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}Inline: "inline" | "default"
+					= "${json.inline ? 'inline' : 'default'}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}Size: "sm" | "md" | "lg" = "${json.size ?? ''}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}InvalidText = "${json.invalidText ?? ''}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}Label = "${json.labelText ?? ''}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}HelperText = "${json.helperText ?? ''}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}WarnText = "${json.warnText ?? ''}";`,
+				outputs: (_) => '',
+				imports: ['SelectModule'],
+				code: ({ json }) => {
+					return `<cds-select
+						[disabled]="${nameStringToVariableString(json.codeContext?.name)}Disabled"
+						[warn]="${nameStringToVariableString(json.codeContext?.name)}Warn"
+						[warnText]="${nameStringToVariableString(json.codeContext?.name)}WarnText"
+						[size]="${nameStringToVariableString(json.codeContext?.name)}Size"
+						[invalid]="${nameStringToVariableString(json.codeContext?.name)}Invalid"
+						[invalidText]="${nameStringToVariableString(json.codeContext?.name)}InvalidText"
+						[label]="${nameStringToVariableString(json.codeContext?.name)}Label"
+						[helperText]="${nameStringToVariableString(json.codeContext?.name)}HelperText"
+						[display]="${nameStringToVariableString(json.codeContext?.name)}Inline"
+						${angularClassNamesFromComponentObj(json)}>
+						${json.items.map((step: any) =>
+							step.items && step.items.length > 0 ? `<optgroup
+							label="${step.label}"
+							${step.disabled ? 'disabled' : ''}>
+							${step.items.map((child: any) => `<option
+								value="${child.value}"
+								${child.disabled ? 'disabled' : ''}
+								${child.hidden ? 'hidden' : ''}>
+									${child.text}
+								</option>`
+							).join('\n')}
+							</optgroup>` : `<option
+							value="${step.value}"
+							${step.disabled ? 'disabled' : ''}
+							${step.hidden ? 'hidden' : ''}>
+								${step.text}
+							</option>`
+						).join('\n')}
```

Comment:
somehow feels like the system should handle this without having to do it explicitly 🤔

Reply:
could you elaborate this a bit more.